Tests/test_file_apng.py

```diff
@@ -689,3 +689,13 @@ def test_different_modes_in_later_frames(mode, default_image, duplicate, tmp_pat
     )
     with Image.open(test_file) as reloaded:
         assert reloaded.mode == mode
+
+
+def test_apng_repeated_seeks_give_correct_info() -> None:
+    # https://github.com/python-pillow/Pillow/issues/7700
+    with Image.open("Tests/images/apng/repeated_seeks_give_correct_info.png") as im:
+        for i in range(5):
```

Comment:
```suggestion
    with Image.open("Tests/images/apng/repeated_seeks_give_correct_info.png") as im:
        for i in range(3):
```
You only need 3 to demonstrate the problem
1. Seek through the image from the initial state.
2. Rewind, loading in `self.rewind_state["info"]`. Seek to the second frame, setting the duration in `self.rewind_state["info"]`
3. Rewind again, but `self.rewind_state["info"]` has been modified now to have the duration of the second frame.